### DISCUSSION TITLE
orderByRelation issue for nested relation two levels down and _max support

### DISCUSSION THREAD

**Discussion by ridhwaans** [prisma/prisma#8177] [Q&A] [answered]
`TVShow` has a *has-many* relation with `Season`, and `Season` has a *has-many* relation with `Episode` 

I am trying to query the list of TVShows and order them by vote average field of every episode across season in every TVShow.    
For example there are 2 TV Shows. A TVShow has 5 Seasons each, and a Season has 10 Episodes each  
if the Episode with the highest `vote_average` key value in TVShow B has a higher `vote_average` than the Episode with the highest `vote_average` in TVShow A, then the query will return TVShow B as first in the list and then TVShow A as second position in the list

this is the query
```ts
const result = await prisma.tVShow.findMany({
    include: { genres: true, production_companies: true, seasons: { include: { episodes: true } }, credits: true, similar: true },
    orderBy: {
      seasons: {
          episodes: {
            _max: {
              vote_average: true
            }
          }
      }
    },
    take: 25
  })
```
Its not working as expected. error message and log returned is 
```
[0] Unknown arg `episodes` in orderBy.seasons.episodes for type SeasonOrderByRelationAggregateInput. Did you mean `_count`? Available args:
[0] type SeasonOrderByRelationAggregateInput {
[0]   _count?: SortOrder
[0]   count?: SortOrder
[0] }
[0]
[0]
[0]     at Document.validate (/node_modules/@prisma/client/runtime/index.js:32626:
```
if I added a second orderBy the error is 
```
[0] Unknown arg `_max` in orderBy.episodes._max for type EpisodeOrderByRelationAggregateInput. Available args:
[0]
[0] type EpisodeOrderByRelationAggregateInput {
[0]   _count?: SortOrder
[0]   count?: SortOrder
[0] }
```
how do I correct the query to make the orderBy work or change to another query?  

below is the prisma schema
```
// package.json
"dependencies": {
    "@prisma/client": "^2.26.0",
    ...   
},
  "devDependencies": {
    "prisma": "^2.26.0",
    ...
}
```

```prisma
// schema.prisma

datasource db {
  provider = "sqlite"
  url      = env("DATABASE_URL")
}

generator client {
  provider = "prisma-client-js"
  previewFeatures = ["referentialActions","orderByRelation"]
}

model TVShow {
  id      				Int      				@id @default(autoincrement())
  type					String
  tmdb_id				Int						@unique
  backdrop_path			String
  genres				Genre[]
  name					String
  overview				String
  popularity			Float
  poster_path			String
  production_companies	ProductionCompany[]
  seasons				Season[]
  tagline				String
  vote_average			Float
  vote_count			Int
  logo_path				String?
  credits				Credit[]				
  similar				Similar[]
  imdb_id				String
}

model Season {
	id      		Int      	@id @default(autoincrement())
	tv_show_tmdb_id	Int
	tv_show			TVShow 		@relation(fields: [tv_show_tmdb_id], references: [tmdb_id], onDelete: Cascade)
	tmdb_id			Int			@unique
	air_date		String
	name			String
	overview		String
	poster_path		String
	season_number	Int
	episodes 		Episode[]
}

model Episode {
	id      				Int      	@id @default(autoincrement())
	type					String
	season_tmdb_id  		Int
	season    				Season    	@relation(fields: [season_tmdb_id], references: [tmdb_id], onDelete: Cascade)
	tmdb_id					Int			@unique
	air_date				String
	episode_number			Int
	name					String
	overview				String
	season_number			Int
	still_path				String
	vote_average			Float
	vote_count				Int
}
```

**Comment by ryands17** [comment, marked as answer]
@ridhwaans 👋 

This currently isn't supported by Prisma. It would be great if you could create a feature request [here](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) explaining order by aggregates for nested relations with the above use case so that we can look into this 🙂

**Reply by ridhwaans** [reply]
https://github.com/prisma/prisma/issues/8190

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
